seed/tests/test_match_existing.py

```diff
@@ -0,0 +1,796 @@
+
+        # Verify no matches exist
+        tls_1_view = TaxLotView.objects.get(state_id=tls_1.id)
+        count_result, no_match_indicator = match_merge_in_cycle(tls_1_view.id, 'TaxLotState')
+        self.assertEqual(count_result, 0)
+        self.assertIsNone(no_match_indicator)
+
+        # Make all those states match
+        TaxLotState.objects.filter(pk__in=[tls_2.id, tls_3.id]).update(
+            jurisdiction_tax_lot_id='123MatchID'
+        )
+
+        # Verify that none of the 3 have been merged
+        self.assertEqual(TaxLot.objects.count(), 3)
+        self.assertEqual(TaxLotState.objects.count(), 3)
+        self.assertEqual(TaxLotView.objects.count(), 3)
+
+        count_result, view_id_result = match_merge_in_cycle(tls_1_view.id, 'TaxLotState')
+        self.assertEqual(count_result, 3)
+
+        # There should only be one TaxLotView which is associated to new, merged -State
+        self.assertEqual(TaxLotView.objects.count(), 1)
+        view = TaxLotView.objects.first()
+        self.assertEqual(view_id_result, view.id)
+        self.assertNotIn(view.state_id, [tls_1.id, tls_2.id, tls_3.id])
+
+        # It will have a -State having city as Philadelphia
+        self.assertEqual(view.state.city, 'Golden')
+
+        # The corresponding log should be a System Match
+        audit_log = TaxLotAuditLog.objects.get(state_id=view.state_id)
+        self.assertEqual(audit_log.name, 'System Match')
+
+    def test_match_merge_in_cycle_ignores_taxlots_with_unpopulated_matching_criteria(self):
+        base_details = {
+            'city': 'Golden',
+            'import_file_id': self.import_file_1.id,
+            'data_state': DATA_STATE_MAPPING,
+            'no_default_data': False,
+        }
+        # Create 3 non-duplicate taxlots with unpopulated matching criteria
+        tls_1 = self.taxlot_state_factory.get_taxlot_state(**base_details)
+
+        base_details['city'] = 'Denver'
+        tls_2 = self.taxlot_state_factory.get_taxlot_state(**base_details)
+
+        base_details['city'] = 'Philadelphia'
+        tls_3 = self.taxlot_state_factory.get_taxlot_state(**base_details)
+
+        self.import_file_1.mapping_done = True
+        self.import_file_1.save()
+        match_buildings(self.import_file_1.id)
+
+        # Verify no match merges happen
+        tls_1_view = TaxLotView.objects.get(state_id=tls_1.id)
+        count_result, no_match_indicator = match_merge_in_cycle(tls_1_view.id, 'TaxLotState')
+        self.assertEqual(count_result, 0)
+        self.assertIsNone(no_match_indicator)
+
+        self.assertEqual(TaxLot.objects.count(), 3)
+        self.assertEqual(TaxLotState.objects.count(), 3)
+        self.assertEqual(TaxLotView.objects.count(), 3)
+
+        state_ids = list(TaxLotView.objects.all().values_list('state_id', flat=True))
+        self.assertCountEqual([tls_1.id, tls_2.id, tls_3.id], state_ids)
+
+
+class TestMatchingExistingViewFullOrgMatching(DataMappingBaseTestCase):
```

Comment:
Note, this method and tests will be heavily changed in year-over-year work. It might not be the most useful to review now.